### ISSUE TITLE
Version 5.1.0 casts boolean to string which leads to validation error

### ISSUE THREAD

**Issue by ledniy** [forge42dev/remix-hook-form#114]
On the new version behaviour for checkbox/boolean is broken. If it's false by default and then you'll check it then it will become a string and will fail validation. Code for reproduction: https://github.com/ledniy/remix-hook-form/commit/1e7aa22cf9d1b590071ce65b381728ac8f3004cd

<img width="819" alt="image" src="https://github.com/user-attachments/assets/894c6558-eaa7-4078-a26b-7c5a41be5a9a">

Temporary workaround: Use version `5.0.4`

**Comment by AlemTuzlak** [comment]
Thank you for this report! will be fixed in the next release! figured out the root cause

**Comment by AlemTuzlak** [comment]
should work now with 5.1.1, please re-open if not the case

**Comment by LydiaF** [comment]
Hi Alem,

I am running into this issue with RR7 (I know there is a new issue that we are waiting on Remix team to fix so am using useForm in the meantime). Form is valid when you don't check the box but the form gets the same error as above when checkbox is checked.

Thanks for this fantastic package!

```ts
import { getValidatedFormData } from "remix-hook-form"
import { zodResolver } from "@hookform/resolvers/zod"
import * as zod from "zod"
import { Form } from 'react-router'
import { useForm } from 'react-hook-form'

const schema = zod.object({
  name: zod.string().min(1),
  email: zod.string().email().min(1).default("test@test.com"),
  test: zod.number(),
  test2: zod.boolean().default(false)
})

type FormData = zod.infer<typeof schema>

export const resolver = zodResolver(schema)

export async function action({ request }: { request: Request }) {
  const { errors, data, receivedValues: defaultValues } =
    await getValidatedFormData<FormData>(request, resolver)

  console.log({ errors, data, defaultValues })

  if (errors) {
    // The keys "errors" and "defaultValues" are picked up automatically by useRemixForm
    return { errors, defaultValues }
  }

  // Do something with the data
  return { data }
}

export default function MyForm() {
  const {
    handleSubmit,
    formState: { errors },
    register,
    watch
  } = useForm<FormData>({
    mode: "onSubmit",
    resolver,
  })
  console.log({ watch: watch })

  return (
    <Form onSubmit={handleSubmit} method="POST">
      <label>
        Test:
        <input
          type="number"
          {...register("test", { valueAsNumber: true })}
        />
        {errors.test && <p>{errors.test.message}</p>}
      </label>
      <label>
        Test2:
        <input type="checkbox" {...register("test")} />
        {errors.test && <p>{errors.test.message}</p>}
      </label>
      <label>
        Name:
        <input type="text" {...register("name")} />
        {errors.name && <p>{errors.name.message}</p>}
      </label>
      <label>
        Email:
        <input type="email" {...register("email")} />
        {errors.email && <p>{errors.email.message}</p>}
      </label>
      <button type="submit">Submit</button>
    </Form>
  )
}
```

<img width="599" alt="image" src="https://github.com/user-attachments/assets/ad8b7bd2-6bb6-4087-b67b-a0a713aff66c">

**Comment by AlemTuzlak** [comment]
I'll check it out as soon as I get remix-hook-form back online with rr7! Thank you!

**Comment by LydiaF** [comment]
Hi Alem, fyi, maybe a similar issue though I am not getting an error. With a value with a zod of `z.string().nullable()`, it is not getting converted to `null`, it is staying as `'null'`. I am on rr 7.0.2. Thanks!
